### PR TITLE
feat!: drop support for node 12

### PR DESCRIPTION
**Pull request by ephys**
Note 12 is going EOL in 23 days, Sequelize 7 won't be released by then. Let's drop support for Node 12 in the upcoming major

### CODE REVIEW

**Comment by ephys** [comment]
This will require removing the required checks for node 12, and adding the ones for node 14

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -27,7 +27,7 @@ jobs:
           cache: yarn
       - name: Install dependencies and build sequelize
         run: yarn install --frozen-lockfile --ignore-engines
-        if: matrix.node-version == 12
+        if: matrix.node-version == 14
```
Comment (WikiRik) [inline .github/workflows/ci.yml:30]: Is this still needed?

File: build.js [hunk 2]
```diff
@@ -79,7 +79,7 @@ async function main() {
       // Adds source mapping
       sourcemap: true,
       // The compiled code should be usable in node v12.22
```
Comment (WikiRik) [inline build.js:81]:
```suggestion
      // The compiled code should be usable in node v14.17
```

**Review by WikiRik — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (ephys) [reply .github/workflows/ci.yml]: Let's find out!

Comment (ephys) [reply .github/workflows/ci.yml]: Looks like it isn't needed anymore

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -219,9 +215,9 @@ jobs:
           name: install-build-artifact-node-${{ matrix.node-version }}
       - name: Extract artifact
         run: tar -xf install-build-node-${{ matrix.node-version }}.tar
-      - name: Install pg-native (Node 12)
+      - name: Install pg-native (Node 14)
         run: yarn add pg-native --ignore-engines
-        if: matrix.native && matrix.node-version == 12
+        if: matrix.native && matrix.node-version == 14
       # This step uses npm instead of yarn to minimize the time needed. See #14171
```
Comment (WikiRik) [inline .github/workflows/ci.yml:221]: Can you check if we need this still?

**Review by WikiRik — COMMENTED** [review]
One more comment, but good to see that we can remove some workarounds now

File: .github/workflows/ci.yml (hunk 3 above)
Comment (ephys) [reply .github/workflows/ci.yml:221]: Looks like it is

**Comment by ephys** [comment]
Ready :) We just need to update the required checks for this branch

**Review by WikiRik — APPROVED** [review]

**Pull request merged**